src/eva/data/mon_data_space.py

```diff
@@ -200,87 +196,256 @@ def subset_coordinate(self, ds, coordinate, requested_subset):
+
+        if dims['ydef'] > 1:
+            y_range = channo if coords['ydef'] == 'Channel' else np.arange(1, dims['ydef']+1)
+
+        if dims['zdef'] > 1:
+            z_range = np.arange(1, dims['zdef']+1)
 
-        return count, penalty, omgnbc_sum, total_sum, omgbc_sum, \
-            omgnbc_sum2, total_sum2, omgbc_sum2, cycle_tm
+        return x_range, y_range, z_range
 
     # ----------------------------------------------------------------------------------------------
+
+    def get_ndims_used(self, dims):
+
+        # Ignore dims with values of 0 or 1
+        ndims = len(dims)
+        for x in range(ndims):
+            if list(dims.values())[x] <= 1:
+                ndims -= 1
+
+        return ndims
+
+    # ----------------------------------------------------------------------------------------------
+
+    def load_dset(self, vars, nvars, coords, darr, dims, ndims_used,
+                  x_range, y_range, z_range, cyc_darr):
+
+        # create dataset from file components
+        rtn_ds = None
+
+        for x in range(0, nvars):
+            if ndims_used == 1:
+                d = {
+                    vars[x]: {"dims": (coords['xdef']), "data": darr[x, :, :]}
+                }
+            if ndims_used == 2:
+                d = {
+                    vars[x]: {"dims": (coords['xdef'], coords['ydef']),
+                              "data": darr[x, :, :]}
+                }
+            if ndims_used == 3:
+                d = {
+                    vars[x]: {"dims": (coords['xdef'], coords['ydef'], coords['zdef']),
+                              "data": darr[x, :, :]}
+                }
+
+            new_ds = Dataset.from_dict(d)
+            rtn_ds = new_ds if rtn_ds is None else rtn_ds.merge(new_ds)
+
+        # tack on 'cycle' as a variable
+        # -----------------------------
+        if ndims_used == 1:
+            new_cyc = Dataset(
+                {
+                    'cycle': ((coords['xdef']), cyc_darr),
+                },
+                coords={coords['xdef']: np.arange(1, dims['xdef']+1)},
+            )
+        if ndims_used == 2:
+            new_cyc = Dataset(
+                {
+                    'cycle': ((coords['xdef'], coords['ydef']), cyc_darr),
+                },
+                coords={coords['xdef']: np.arange(1, dims['xdef']+1),
+                        coords['ydef']: np.arange(1, dims['ydef']+1)},
+            )
+        if ndims_used == 3:
+            new_cyc = Dataset(
+                {
+                    'cycle': ((coords['xdef'], coords['ydef'], coords['zdef']), cyc_darr),
+                },
+                coords={coords['xdef']: np.arange(1, dims['xdef']+1),
+                        coords['ydef']: np.arange(1, dims['ydef']+1),
+                        coords['zdef']: np.arange(1, dims['zdef']+1)},
+            )
+
+        rtn_ds = rtn_ds.merge(new_cyc)
+        return rtn_ds
```

Comment:
Your logic is not wrong at all here, but I think I found a way to make this a bit more efficient. I used the factory-like method similar to what I suggested last review. It gets rid of a all of the conditionals which are slow in Python. Please double check to make sure this works:

```
ndims_dict = {
    1: {
        'dims': (coords['xdef']),
        'cycle': Dataset({'cycle': ((coords['xdef']), cyc_darr),},
                         coords={coords['xdef']: np.arange(1, dims['xdef']+1)},)
    },
    2: {
        'dims': (coords['xdef'], coords['ydef']),
        'cycle': Dataset({'cycle': ((coords['xdef'], coords['ydef']), cyc_darr),},
                         coords={coords['xdef']: np.arange(1, dims['xdef']+1),
                         coords['ydef']: np.arange(1, dims['ydef']+1)},)
    },
    3: {
        'dims': (coords['xdef'], coords['ydef'], coords['zdef']),
        'cycle': Dataset({'cycle': ((coords['xdef'], coords['ydef'], coords['zdef']), cyc_darr),},
                         coords={coords['xdef']: np.arange(1, dims['xdef']+1),
                         coords['ydef']: np.arange(1, dims['ydef']+1),
                         coords['zdef']: np.arange(1, dims['zdef']+1)},)
    }
}

# Create empty dict
d = {}

# Loop through vars and fill empty dict
for x in range(0, nvars):
    d[vars[x]] = {
        'dims': ndims_dict[ndims_used]['dims'],
        'data': darr[x, :, :]
    }

# Create Dataset from dictionary
rtn_ds = Dataset.from_dict(d)

# Merge cycle Dataset based on ndims_used
rtn_ds = rtn_ds.merge(ndims_dict[ndims_used]['cycle'])
```

Reply:
Cool.  I had tried doing that here before but, looking at your code, I see where I might have gone wrong in that attempt.  I'll give it another go.

Reply:
@kevindougherty-noaa I'm getting this error with the 'cycle' definition in the factory-like construct:

`ValueError: Could not convert tuple of form (dims, data[, attrs, 
encoding]): ('Channel', array([[[datetime.datetime(2023, 4, 9, 0, 0), .... to Variable`

Any thoughts on how to correct that?

Reply:
@EdwardSafford-NOAA I am not too sure what you mean. Is it in issue in the dictionary or when you try and merge?

Reply:
It's in the dictionary.  I've tried paring down/simplifying the dictionary structure.  The 'dims' definition is fine.  The 'cycle' is the source of the problem.

Reply:
Maybe `Dataset()` can't work in the dictionary. I unfortunately don't have a solution for the error you are getting.

Reply:
I worked my way around the `Dataset `issue.  This isn't quite as concise as what you'd hoped for but all the the `if/elif/else `on `num_dims `is now eliminated.  Please see what you think.